pkg/controller/pod.go

```diff
@@ -202,15 +214,87 @@ func (c *Controller) syncPod(ctx context.Context, pod *corev1.Pod) error {
 		}
 	}
 
```

Comment:
we may need to add a lock here. since there are multiple `read -> process -> write`. these write might overwrite another write by another coroutine.

Reply:
+1 on #1. It seems very coarse to cache a pod IP for 5 minutes.

Reply:
We refresh the Cache every 5 mins with what we have in CloudMap. So, if for whatever reason we didn't register a Pod to a cloudmap service, Cache will not have that pod when we refresh it from cloudmap at the end of 5 min window. 

I don't see the need to clean the cache on deregister instance here. It just stays there and gets wiped out once cache TTL expires. We will not be using it in anyform. Isn't it harmless?

-Efe, it is a TTL based cache so we picked a 5 min window.